File: app/src/main/java/com/example/myapplication/preferences/PreferencesManager.java
package com.example.myapplication.preferences;

import android.app.Application;
import android.content.Context;
import android.content.SharedPreferences;

import androidx.lifecycle.LiveData;

import com.example.myapplication.model.Animal;

public class PreferencesManager {

    private static LiveData<Animal> animal;
    public static final String SHARED_PREFENCES_NAME = "preferences";
    public static final String SHARED_PREFENCES_KEY_ANIMAL_NAME = "animal_name";
    private static SharedPreferences sharedPreferences;

    public PreferencesManager(Application application) {
        if (sharedPreferences == null){
            sharedPreferences = application.getSharedPreferences(SHARED_PREFENCES_NAME,Context.MODE_PRIVATE);
        }
    }
    public void saveAnimal(Animal animal){
        sharedPreferences.edit().putString(SHARED_PREFENCES_KEY_ANIMAL_NAME, animal.getName()).apply();
    }
    public Animal getAnimal(){
        return new Animal(sharedPreferences.getString(SHARED_PREFENCES_KEY_ANIMAL_NAME,"Trống")) ;
    }

}

File: app/src/main/java/com/example/myapplication/preferences/PreferencesManager.kt
package com.example.myapplication.preferences

import android.app.Application
import android.content.Context
import android.content.SharedPreferences
import com.example.myapplication.model.Animal

class PreferencesManager(application: Application) {
    val SHARED_PREFENCES_NAME = "preferences"
    val SHARED_PREFENCES_KEY_ANIMAL_NAME = "animal_name"
    private var sharedPreferences: SharedPreferences? = null

    init {
        if (sharedPreferences == null) {
            sharedPreferences = application.getSharedPreferences(SHARED_PREFENCES_NAME, Context.MODE_PRIVATE)
        }
    }

    fun saveAnimal(animal: Animal) {
        sharedPreferences!!.edit().putString(SHARED_PREFENCES_KEY_ANIMAL_NAME, animal.name).apply()
    }

    fun getAnimal(): Animal {
        return Animal(sharedPreferences!!.getString(SHARED_PREFENCES_KEY_ANIMAL_NAME,"Trống")!!)
    }
}